javav2/example_code/dynamodb/src/main/java/com/example/enhanced/dynamodb/QueryRecordsWithFilter.java

```diff
@@ -0,0 +1,195 @@
+                            .getter(Record::getId)
+                            .setter(Record::setId)
+                            .tags(primaryPartitionKey()))
+                    .addAttribute(String.class, a -> a.name("sort")
+                            .getter(Record::getSort)
+                            .setter(Record::setSort)
+                            .tags(primarySortKey()))
+                    .addAttribute(String.class, a -> a.name("attribute")
+                            .getter(Record::getAttribute)
+                            .setter(Record::setAttribute))
+                    .addAttribute(String.class, a -> a.name("attribute2*")
+                            .getter(Record::getAttribute2)
+                            .setter(Record::setAttribute2)
+                            .tags(secondaryPartitionKey("gsi_1")))
+                    .addAttribute(String.class, a -> a.name("attribute3")
+                            .getter(Record::getAttribute3)
+                            .setter(Record::setAttribute3)
+                            .tags(secondarySortKey("gsi_1")))
+                    .build();
+
+    // QUery the Record table
+    public static void main(String[] args) {
+
+        //Create a DynamoDbClient object
+        Region region = Region.US_EAST_1;
+        DynamoDbClient ddb = DynamoDbClient.builder()
+                .region(region)
+                .build();
+        try{
+            // snippet-start:[dynamodb.java2.enhanced.queryfilter.main]
+            // Create a DynamoDbEnhancedClient and use the DynamoDbClient object
+            DynamoDbEnhancedClient enhancedClient = DynamoDbEnhancedClient.builder()
+                    .dynamoDbClient(ddb)
+                    .build();
+
+            //Create a DynamoDbTable object
+            DynamoDbTable<Record> mappedTable = enhancedClient.table("Record", TABLE_SCHEMA);
+
+            //Get the row where Attribute3 is 20
+            AttributeValue att = AttributeValue.builder()
+                    .s("20")
+                    .build();
+
+            Map<String, AttributeValue> expressionValues = new HashMap<>();
+            expressionValues.put(":value", att);
+
+            Expression expression = Expression.builder()
+                    .expression("attribute3 = :value")
+                    .expressionValues(expressionValues)
+                    .build();
+
+            // Create a QueryConditional object that is used in the query operation
+            QueryConditional queryConditional = QueryConditional
+                    .keyEqualTo(Key.builder().partitionValue("id-value")
+                    .build());
+
+            // Get items in the Record table and write out the ID value
+            Iterator<Page<Record>> results = mappedTable.query(QueryEnhancedRequest.builder()
+                    .queryConditional(queryConditional)
+                    .filterExpression(expression)
+                    .build()).iterator();
```

Comment:
Use items().iterator() instead of just iterator() then you don't have to paginate.

Reply:
done